httpclient/httpclient.go

```diff
@@ -491,9 +491,9 @@ func (c *Client) retryRequest(ctx context.Context, name string, r Request, newRe
 			r.headerFn(res.Header)
 		}
 
-		err = r.decodeBody(res, true, attemptCounter)
-		if err != nil {
-			return err
+		errDecode := r.decodeBody(res, true, attemptCounter)
```

Comment:
Is this part making any functional change? It's not leaping out at me

Reply:
We seem to also do this on L482 so that it's also not added to the span

Reply:
I don't see how this would stop this being added to the span, the error is still returned, and `err` is a named return parameter